src/matchlight/alert.py

```
@@ -0,0 +1,284 @@
+                'record_id_filter': record_id,
+                'mtime': mtime,
+                'offset': offset
+            }
+        )
+        alerts = []
+        for payload in response.json().get('alerts', []):
+            alerts.append(Alert.from_mapping(payload))
+        return alerts
+
+    def edit(self, alert_id, seen=None, archived=None):
+        """Edits an alert.
+
+        Example:
+            Archive an alert::
+
+                >>> alert
+                <Alert(number=1024,
+                id="0760570a2c4a4ea68d526f58bab46cbd")>
+                >>> ml.alert.edit(alert, archived=True)
+                {
+                    'seen': True,
+                    'archived': True
+                }
+
+        Arguments:
+            alert (:obj:`str`): An alert id.
+            seen (:obj:`bool`, optional):
+            archived (:obj:`bool`, optional):
+
+        Returns:
+            :obj:`dict`: Updated alert metadata.
+
+        """
+        data = {}
+        if seen is not None:
+            data['seen'] = seen
+
+        if archived is not None:
+            data['archived'] = archived
+
+        response = self.conn.request(
+            '/alert/{}/edit'.format(alert_id),
+            data=json.dumps(data)
+        )
+        response = response.json()
+        return {
+            'seen': response['seen'],
+            'archived': response['archived']
+        }
+
+    def get_details(self, alert_id):
+        """Returns details of an alert by the given alert ID.
+
+        Args:
+            alert_id (:obj:`str`): The alert identifier.
+
+        Returns:
+           :obj:`dict`: map of the alert details.
+
+        """
+        if isinstance(alert_id, Alert):
+            alert_id = alert_id.id
+
+        try:
+            response = self.conn.request('/alert/{}/details'.format(alert_id))
+            return response.json()
+        except matchlight.error.APIError as err:
+            if err.args[0] == 404:
+                return
```

So if a user tries to get details on an alert that does not exists (404), we silently fail. Why? Do we do anything like this elsewhere in the code for 404s?